packages/common-library/src/common_library/pydantic_settings_validators.py

```diff
@@ -0,0 +1,20 @@
+import datetime
+
+from pydantic import field_validator
+
+
+def _get_float_string_as_seconds(
```

Comment:
The function name is really unclear, and it doesn't clearly convey the intended behavior. Based on the code, the input-output mapping appears to work as follows:

- For `string` inputs: it attempts to convert to `float`; if conversion fails, it returns the original `string`.
- For `float` inputs: it returns the `float` unchanged (bypass).
- For `timedelta` inputs: it returns the `timedelta` unchanged (bypass).